src/main/java/learnositysdk/request/Init.java

```diff
@@ -348,63 +351,67 @@ private void setServiceOptions() throws Exception
      */
     private void validateRequiredArgs(String service, Object securityPacket, String secret) throws Exception
     {
-        
         if (service.isEmpty()) {
             throw new Exception("The `service` argument wasn't found or was empty");
         } else if (!Arrays.asList(this.validServices).contains(service.toLowerCase())) {
             throw new Exception("The service provided " + service + " is not valid");
         }
         this.service = service;
+        this.canAlterRequestData = !service.equals("data");
 
-        // In case the user gave us a securityPacket String, convert to a JSONOBject
-        
-
+        // In case the user gave us a securityPacket String, convert to a JSONObject
         this.validateSecurityPacket(securityPacket);
-        
+
         if (secret.isEmpty()) {
             throw new Exception("The `secret` argument must be a valid string");
         }
         this.secret = secret;
     }
 
     /**
-     * Validates the request packet argument
-     * @param requestPacket
-     * @throws Exception
+     * Sets the request packet, updating the stored string version as well
+     *
+     * @param requestPacket The request packet to store
      */
-    private void validateRequestPacket(Object requestPacket) throws Exception
+    private void setRequestPacket(Object requestPacket) throws Exception
     {
-        if (requestPacket instanceof JSONObject) {
-            this.requestPacket = new JSONObject(requestPacket.toString());
-            this.requestString = requestPacket.toString();
-        } else {
-            if (requestPacket instanceof String) {
-                this.requestPacket = new JSONObject((String)requestPacket);
-                this.requestString = (String)requestPacket;
-            } else if (requestPacket instanceof Map) {
-                this.requestPacket = new JSONObject((Map)requestPacket);
-                this.requestString = this.requestPacket.toString();
+        JSONObject convertedPacket = validateRequestPacket(requestPacket);
+
+        // Prevent updating request packet in situations where we must not do so
```

Comment:
I'm not especially happy with this approach, but we need to prevent all altering the request packet for Data API requests (including any conversion from JSONObject back to string which could potentially alter the order of the keys) because we don't return the new packet - only the Security object. This has been added into our backlog to be reviewed during feature parity audit for the SDKs.

Reply:
Ah, see my comment above https://github.com/Learnosity/learnosity-sdk-java/pull/15/commits/6cce99e245e3b11ec6dc1732b22f2f9021891d12#r279607096

Reply:
RESOLVED: We've cleaned this up a little bit with the new approach in the polish commit